### PR TITLE
Show matched target when creating the match

### PR DESCRIPTION
**Pull request by ManuViola77**
This pr implements the user story [Show Matched Target](https://trello.com/c/B3I295Sl/12-5-as-a-user-i-should-be-able-to-start-a-conversation-with-any-of-my-compatible-targets-so-i-can-talk-about-our-common-interests)

When a user creates a target that matches another existing target, it shows a modal with the matched information and gives the option to go right to the chat and start talking.

### CODE REVIEW

File: src/constants/shapes.js [hunk 1]
```diff
@@ -53,3 +55,14 @@ export const topicItemShape = shape({
...
+
+export const userShape = shape({
+  avatar: avatarShape,
+  email: string,
+  firstName: string,
+  fullName: string,
+  gender: string,
+  id: number,
+  lastName: string,
+  username: string,
```
Comment (tamarasuarezrod) [inline src/constants/shapes.js]:
```suggestion
  username: string.isRequired,
```
remember you can do this if the value is required for the app to not crash

File: src/reducers/targetsReducer.js [hunk 2]
```diff
@@ -1,9 +1,29 @@
...
 const initialState = { targetsList: [] };
 
 const targetsReducer = {
+  [createTargetReset]: (store, action) => {
+    store.matchedUser = undefined;
+    store.matchId = undefined;
+  },
+
+  [createTargetSuccess]: (store, action) => {
+    if (action.payload.matchConversation) {
```
Comment (tamarasuarezrod) [inline src/reducers/targetsReducer.js]: destructure the action outside the if so you can use it here
```suggestion
    if (matchConversation) {
```

File: src/components/MatchTargetModal/index.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    } = matchedUser;
+
+    const chatParam = { matchId, user: matchedUser };
+
+    const goToChat = () => {
+      closeModal();
+      navigation.push(CHAT_SCREEN, { [targetMatchParam]: chatParam });
+    };
+
+    console.log('matchedUser: ', matchedUser);
```
Comment (tamarasuarezrod) [inline src/components/MatchTargetModal/index.js]: delete this 😄

Comment (ManuViola77) [reply src/components/MatchTargetModal/index.js]: this was made to see the user's shape, I forgot to delete it

**Pull request merged**